### ISSUE TITLE
Can't mass-assign protected attributes for Audited::Adapters::ActiveRecord::Audit: action, audited_changes, comment

### ISSUE THREAD

**Issue by davingee** [collectiveidea/audited#186]
Hello, 
Getting a Can't mass-assign protected attributes error

Can't mass-assign protected attributes for Audited::Adapters::ActiveRecord::Audit: action, audited_changes, comment

This is happening when I the_model.update_attributes(params[:the_model])

if I stick this in the model I am saving it works    Audited::Adapters::ActiveRecord::Audit.send(:attr_accessible, :action, :audited_changes, :comment)

Any idea on where to start?

**Comment by kennethkalmer** [comment]
Are you updating instances of the Audit model? Could we have some sample code to make sure we understand what is going on?

To my mind we should protect the `action`, `audited_changes` and `associated` attributes, otherwise malicious users can tamper with these values and leave the audits useless...

**Comment by kennethkalmer** [comment]
And I noticed now we have [a section in the README](https://github.com/collectiveidea/audited#using-attr_protected-or-strong_parameters) discussing this issue.

**Comment by yatsura** [comment]
I can't speak to what problems other people are having but I've currently upgrading an application to Rail 4.1 from 3.2.  This means that it has `attr_accessible` in the models so I have the protected_attributes gem enabled in Gemfile.

I've create a small test application to confirm that it is the protected_attributes gem it also needs
`config.active_record.mass_assignment_sanitizer = :strict` in the environments/test.rb file

If I change auditor.rb line 206 to

`run_callbacks(:audit)  { self.audits.create(attrs, without_protection: true) } if auditing_enabled`

My tests pass but I've not thought through what problems this might cause...

**Comment by yatsura** [comment]
Also should say that `:allow_mass_assignment` does not fix this issue.  I've create a rails 4.1 app than demos this. https://github.com/yatsura/audited_168

**Comment by mordocai** [comment]
@yatsura Did you ever figure out anything here? I've got the exact same setup as you (moving from 3.2 to 4 and having protected_attributes enabled). I haven't changed any audited code yet like you did, and haven't found any solutions that work.

**Comment by yatsura** [comment]
@mordocal Honestly I had spent too much time and so we decided to rewrite to strong_parameters.  We didn't have a huge amount of models in the application.  I meant to get back to this in my own time...

**Comment by mordocai** [comment]
@yatsura Yeah, we're moving that direction as well. It'll take a good amount of work but seems less painful than dealing with this.

**Comment by y00rb** [comment]
@Eric-Guo   @JunichiIto 
:+1:

**Comment by pvijayror** [comment]
uknow attribute in rails 4.2  "attr_accessible"

**Comment by Eric-Guo** [comment]
@pvijayror [no more attr_accessible in rails 4](http://rubyjunction.us/no-more-attr-accessible-in-ruby-on-rails-4), maybe you need [protected_attributes](http://rubygems.org/gems/protected_attributes)?
